docs/architecture/architecture.md

```diff
@@ -14,6 +14,91 @@ built to be multi-tenant.
 | [Private SaaS (Customer VPC)](./deployment#private-saas-customer-vpc) | Enterprises in the Cloud                                        |
 | [On-Premise](./deployment#on-premise-deployment)                      | Large Enterprises in the middle of cloud migration (rare cases) |
 
+## High-Level Architecture
+
+There are four components of a successful Prophecy deployment:
+
+- **Prophecy IDE** - The development environment, including Prophecy Microservices and Cloud Infrastructure, which is deployed using one of these options: [Public SaaS](#public-saas), [Private SaaS](#private-saas-customer-vpc), or (rarely) [On-Premise](#on-premise-deployment).
```

Comment:
We dont want to highlight on-premise much   ( Good to check with Product team once)

Reply:
true